src/core/generator/generator.rs

```diff
@@ -34,219 +28,287 @@ fn resolve_file_descriptor_set(descriptor_set: FileDescriptorSet) -> Result<File
     Ok(descriptor_set)
 }
 
-// TODO: move this logic to ResourceReader.
-async fn fetch_response(
-    url: &str,
-    runtime: &TargetRuntime,
-) -> anyhow::Result<RequestSample> {
-    let parsed_url = Url::parse(url)?;
-    let request = reqwest::Request::new(Method::GET, parsed_url.clone());
-    let resp = runtime.http.execute(request).await?;
-    let body = serde_json::from_slice(&resp.body)?;
-    Ok(RequestSample::new(parsed_url, body))
+pub struct ConfigInput {
+    pub schema: String,
+    pub source: config::Source,
 }
 
+pub struct JsonInput {
+    pub url: Url,
+    pub data: Value,
+}
+
+pub struct ProtoInput {
+    pub data: ProtoMetadata,
+}
+
+/// Generator offers an abstraction over the actual config generators and allows
+/// to generate the single config from multiple sources. i.e (Protobuf and Json)
 pub struct Generator {
-    runtime: TargetRuntime,
-    reader: ResourceReader<Cached>,
-    proto_reader: ProtoReader,
+    operation_name: Option<String>,
+    proto_samples: Option<Vec<ProtoInput>>,
+    json_samples: Option<Vec<JsonInput>>,
+    config_samples: Option<Vec<ConfigInput>>,
```

Comment:
Config is not a sample. Rename the field.

Reply:
Create a `Vec<Input>` where `Input` is an Enum. Keep proto, json, config all as inputs.